.eslintrc.js

```diff
@@ -1,132 +1,107 @@
-    'app/vendor/**',
-    'test/e2e/send-eth-with-private-key-test/**',
-    'nyc_output/**',
-    '.vscode/**',
-    'lavamoat/*/policy.json',
-    'storybook-build/**',
-  ],
-
-  extends: ['@metamask/eslint-config', '@metamask/eslint-config-nodejs'],
-
-  plugins: ['@babel', 'import', 'jsdoc'],
-
-  globals: {
-    document: 'readonly',
-    window: 'readonly',
-  },
-
-  rules: {
-    'default-param-last': 'off',
-    'prefer-object-spread': 'error',
-    'require-atomic-updates': 'off',
+  // Ignore files which are also in .prettierignore
+  ignorePatterns: ['app/vendor/**', 'dist/**/*', 'development/chromereload.js'],
+  overrides: [
+    /**
+     * == Modules ==
+     *
+     * The first two sections here, which cover module syntax, are mutually
+     * exclusive: the set of files covered between them may NOT overlap. This is
+     * because we do not allow a file to use two different styles for specifying
+     * imports and exports (however theoretically possible it may be).
+     */
 
-    // This is the same as our default config, but for the noted exceptions
-    'spaced-comment': [
-      'error',
-      'always',
-      {
-        markers: [
-          'global',
-          'globals',
-          'eslint',
-          'eslint-disable',
-          '*package',
-          '!',
-          ',',
-          // Local additions
-          '/:', // This is for our code fences
-        ],
-        exceptions: ['=', '-'],
+    {
+      /**
+       * Modules (CommonJS module syntax)
+       *
+       * This is code that uses `require()` and `module.exports` to import and
+       * export other modules.
+       */
+      files: [
+        '.eslintrc.js',
+        '.eslintrc.*.js',
+        '.mocharc.js',
+        '*.config.js',
+        'development/**/*.js',
+        'test/e2e/**/*.js',
+        'test/helpers/*.js',
+        'test/lib/wait-until-called.js',
+      ],
+      extends: [
+        path.resolve(__dirname, '.eslintrc.base.js'),
+        path.resolve(__dirname, '.eslintrc.node.js'),
+        path.resolve(__dirname, '.eslintrc.babel.js'),
+      ],
+      parserOptions: {
+        sourceType: 'module',
       },
-    ],
-
-    'import/no-unassigned-import': 'off',
-
-    'no-invalid-this': 'off',
-    '@babel/no-invalid-this': 'error',
-
-    // Prettier handles this
-    '@babel/semi': 'off',
-
-    'node/no-process-env': 'off',
-
-    // Allow tag `jest-environment` to work around Jest bug
-    // See: https://github.com/facebook/jest/issues/7780
-    'jsdoc/check-tag-names': ['error', { definedTags: ['jest-environment'] }],
-
-    // TODO: remove this override
-    'padding-line-between-statements': [
-      'error',
-      {
-        blankLine: 'always',
-        prev: 'directive',
-        next: '*',
+      rules: {
+        // This rule does not work with CommonJS modules. We will just have to
+        // trust that all of the files specified above are indeed modules.
+        'import/unambiguous': 'off',
```

Comment:
In refactoring this config, I discovered that ESLint was confused over whether this file was a script or a module. It seemed that this was not possible because I thought that [we'd configured `eslint-plugin-import` to work with CommonJS](https://github.com/MetaMask/eslint-config/blob/ec27156931a0cda1b34ff46937b8296742dcb1c0/packages/base/src/index.js#L278). That is true, but that setting only applies to the `no-unresolved` rule. The `no-ambiguous` rule is separate and [only seems to work on ES module files](https://github.com/import-js/eslint-plugin-import/blob/41d4500d83e969a0b24be837625f754b9eadada6/utils/unambiguous.js) . As far as I can tell, this is a long-standing bug and not a feature. In the meantime, we just have to turn this rule off for CommonJS files.